### PR TITLE
Add example code referenced in issue #452 

### PR DESCRIPTION
**Pull request by ajpowelsnl**
This PR adds example code from Compiler Explorer to `MinLoc` built-in reducer.

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
The indentation of the entire code block does not look right

**Comment by ajpowelsnl** [comment]
> The indentation of the entire code block does not look right

Fixed indentation (w/ spaces), so it now looks like example in `View`.

File: docs/source/API/core/builtinreducers/MinLoc.rst [hunk 1]
```diff
@@ -108,3 +108,44 @@ Additional Information
...
+            val.val = a(i,j,k);
+            val.loc[0] = i;
+            val.loc[1] = j;
+            val.loc[2] = k;
+        }
+        },MinLoc_t(result));
+        printf("%lf %i %i %i\n",result.val,result.loc[0],result.loc[1],result.loc[2]);
+        }
+        Kokkos::finalize();
+    }
```
Comment (masterleinad) [inline docs/source/API/core/builtinreducers/MinLoc.rst:151]: `clang-format-8` gives
```C++
#include <Kokkos_Core.hpp>
struct Idx3D_t {
  int value[3];
  int& operator[](int i) { return value[i]; }
  const int& operator[](int i) const { return value[i]; }
};
template <>
struct Kokkos::reduction_identity<Idx3D_t> {
  static constexpr Idx3D_t min() { return {0, 0, 0}; }
};
int main(int argc, char* argv[]) {
  Kokkos::initialize(argc, argv);
  {
    Kokkos::View<double***> a("A", 5, 5, 5);
    Kokkos::deep_copy(a, 10);
    a(2, 3, 1)        = 5;
    using MinLoc_t    = Kokkos::MinLoc<double, Idx3D_t>;
    using MinLocVal_t = typename MinLoc_t::value_type;
    MinLocVal_t result;
    Kokkos::parallel_reduce(
        Kokkos::MDRangePolicy<Kokkos::Rank<3>>({0, 0, 0}, {5, 5, 5}),
        KOKKOS_LAMBDA(int i, int j, int k, MinLocVal_t& val) {
          if (a(i, j, k) < val.val) {
            val.val    = a(i, j, k);
            val.loc[0] = i;
            val.loc[1] = j;
            val.loc[2] = k;
          }
        },
        MinLoc_t(result));
    printf("%lf %i %i %i\n", result.val, result.loc[0], result.loc[1],
           result.loc[2]);
  }
  Kokkos::finalize();
}
```
(without the preceding whitespace).

Comment (ajpowelsnl) [reply docs/source/API/core/builtinreducers/MinLoc.rst:151]: So I tried your clang-formatted version (above), but it did not render correctly, and had many warnings (e.g., "WARNING: Block quote ends without a blank line; unexpected unindent").  It is possible that we do not have our project set up correctly / completely for C++ (see [#451](https://github.com/kokkos/kokkos-core-wiki/pull/451)).  Example formatting can be a topic for tomorrow.

Comment (nmm0) [reply docs/source/API/core/builtinreducers/MinLoc.rst:151]: We need to tab over the output of clang format, directives need to be tabbed

File: docs/source/API/core/builtinreducers/MinLoc.rst [hunk 2]
```diff
@@ -108,3 +108,64 @@ Additional Information
 * Requires: ``Index`` has ``operator =`` defined. ``Kokkos::reduction_identity<Index>::min()`` is a valid expression.
 
 * In order to use MinLoc with a custom type of either ``Scalar`` or ``Index``, a template specialization of ``Kokkos::reduction_identity<CustomType>`` must be defined. See `Built-In Reducers with Custom Scalar Types <../../../ProgrammingGuide/Custom-Reductions-Built-In-Reducers-with-Custom-Scalar-Types.html>`_ for details
+
+Example
+-------
+
+.. code-block:: cpp
+
+	#include <Kokkos_Core.hpp>
```
Comment (masterleinad) [inline docs/source/API/core/builtinreducers/MinLoc.rst:117]: Can you use for whitespaces instead of one tab?

Comment (ajpowelsnl) [reply docs/source/API/core/builtinreducers/MinLoc.rst:117]: sure.  I'll have to do it by hand, though.

Comment (masterleinad) [reply docs/source/API/core/builtinreducers/MinLoc.rst:117]: Something like `%s/\t/    /g` in `vi`.

**Review by masterleinad — APPROVED** [review]
Fine with me.

**Pull request merged**